.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: requirements-txt-fixer
+
+  - repo: https://github.com/mgedmin/check-manifest
+    rev: "0.49"
+    hooks:
+      - id: check-manifest
+        stages: [manual]
+
+  - repo: local
+    hooks:
+      - id: prettier-staged
+        name: Prettier Staged
+        entry: git-format-staged -f 'prettier --ignore-unknown --stdin --stdin-filepath "$FILENAME"'
```

Comment:
Does this require each contributor to install git-format-staged and prettier on their local machine in advance?

Reply:
It is included in the package.json of the root directory so it gets installed from there

Reply:
Should I mention it in the README under Contributing?

Reply:
Hi，I would like to confirm, after having this PR, if the developer clones the repository, can he directly git commit the code without performing any additional operations?

Reply:
I added installation command for it in the latest commit

Reply:
Yes, with the update he can